enterprise/server/remote_execution/runner/runner.go

```diff
@@ -1048,6 +1069,10 @@ func (p *pool) take(ctx context.Context, key *rnpb.RunnerKey) (*commandRunner, e
 	p.mu.Lock()
 	defer p.mu.Unlock()
 
+	if p.isShuttingDown {
```

Comment:
This line should fix the panic. The other changes are optimizations so that we exit a little earlier on shut down

Reply:
Looking into this more, I actually don't think this is the fix. This is just another optimization so we shut down earlier on shut down